### PR TITLE
fix: Moved "Return to Regular Route" into root directions list

### PR DESCRIPTION
**Pull request by hannahpurcell**
The little things I like about this approach is that
- "Return to Regular Route" is put with other directions, which I think it should be!
- The direction is in the root hook instead of scattered in multiple components
- The test for `diversionPage` now sufficiently covers the check "was 'Return to Regular Route' added to the directions sequence"
- We can still choose to make directions more structured with typing later (per Josh's suggestion)

Also included, updating the Storybook story for this!

### CODE REVIEW

File: assets/src/hooks/useDetour.ts [hunk 1]
```diff
@@ -88,7 +88,11 @@ export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
   )
 
   const coordinates = isOk(detourShape) ? detourShape.ok.coordinates : []
-  const directions = isOk(detourShape) ? detourShape.ok.directions : undefined
+  const directions = isOk(detourShape)
+    ? detourShape.ok.directions?.concat({
+        instruction: "Return to Regular Route",
```
Comment (firestack) [inline assets/src/hooks/useDetour.ts:93]: question: this text doesn't match the copy in the Figma, was it updated somewhere else that I missed?

File: assets/src/hooks/useDetour.ts [hunk 2]
```diff
@@ -88,7 +88,11 @@ export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
...
+      })
+    : undefined
```
[the remark this answers is not in the document]
Comment (hannahpurcell) [reply assets/src/hooks/useDetour.ts]: Ahhh yes! Thank you

File: assets/src/hooks/useDetour.ts (hunk 1 above)
Comment (hannahpurcell) [reply assets/src/hooks/useDetour.ts]: Ohhhhh!! I got so locked onto the ticket title that I glossed over the diff with the figma

File: assets/src/hooks/useDetour.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshlarson) [reply assets/src/hooks/useDetour.ts]: This is enough of a tricky edge case that I would recommend having a test for it. Ideally at the `diversionPage` level (to protect against refactors), but I would be okay with the `useDetour` level as well or instead.

[the remark this answers is not in the document]
Comment (hannahpurcell) [reply assets/src/hooks/useDetour.ts]: Paired with @firestack to write the tests, and I understand things so much better! (Thanks, Kayla 😄 )

Added a test at the `diversionPage` level to verify the text appears at the right times. (Is the second non-existence test too large?) I was not able to include a test to verify the `<strong>` attribute. I think [this release](https://github.com/testing-library/dom-testing-library/pull/1269) of `testing-library` could resolve how `findByRole` ignores the `strong` role.

File: assets/tests/components/detours/diversionPage.test.tsx [hunk 3]
```diff
@@ -242,6 +242,70 @@ describe("DiversionPage", () => {
...
+
+    expect(screen.queryByText("Regular Route")).toBeNull()
+
+    await act(async () => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    await act(async () => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
```
Comment (firestack) [inline assets/tests/components/detours/diversionPage.test.tsx:302]: suggestion: I don't think the inner blocks need `async`?
```suggestion
    await act(() => {
      fireEvent.click(originalRouteShape.get(container))
    })

    await act(() => {
      fireEvent.click(originalRouteShape.get(container))
    })
```

**Review by firestack — DISMISSED** [review]
Two comments but otherwise this looks good!

File: assets/tests/components/detours/diversionPage.test.tsx [hunk 4]
```diff
@@ -242,6 +242,70 @@ describe("DiversionPage", () => {
...
+
+    fireEvent.click(screen.getByRole("button", { name: "Undo" }))
+
+    expect(screen.queryByText("Regular Route")).toBeNull()
```
[the remark this answers is not in the document]
Comment (hannahpurcell) [reply assets/tests/components/detours/diversionPage.test.tsx:306]: Ok! What are the things you think about when choosing between those options? And why choose one over the other here? I saw a lot of `.toBeNull()` matchers in this test suite for elements that should not be present after some actions, but I've also seen `.not.toBeInTheDocument()`

[the remark this answers is not in the document]
Comment (firestack) [reply assets/tests/components/detours/diversionPage.test.tsx:306]: AFAIK the `toBeNull` matchers on DOM elements are just legacy code. I prefer to use the jest-dom matchers wherever possible when working with DOM elements, just so that the code both reads better and is clearer about what it's asserting.

File: assets/tests/components/detours/diversionPage.test.tsx [hunk 5]
```diff
@@ -242,6 +242,70 @@ describe("DiversionPage", () => {
...
+    })
+
+    await act(async () => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    expect(await screen.findByText("Regular Route")).toBeVisible()
+  })
+
+  test("does not show 'Regular Route' when detour not finished", async () => {
```
Comment (joshlarson) [inline assets/tests/components/detours/diversionPage.test.tsx:275]:
```suggestion
  test("does not show 'Regular Route' when detour is not finished", async () => {
```

Nit: grammar

File: assets/tests/components/detours/diversionPage.test.tsx (hunk 3 above)
Comment (firestack) [reply assets/tests/components/detours/diversionPage.test.tsx:302]: No because `act` is `async`

Comment (hannahpurcell) [reply assets/tests/components/detours/diversionPage.test.tsx:302]: Ah that makes sense! Had to [read up](https://medium.com/@bmb21/reacts-sync-and-async-act-caa297b658b0) a tad on act async, and it does work like it sounds. (There are some other functions here that do use `async` to wrap an identical click action, like `missed stops are filled in when detour is complete`, so I initially was copying those. Seems like `async` is unneeded there, too)

Comment (firestack) [reply assets/tests/components/detours/diversionPage.test.tsx:302]: ^I'm confirming this assumption though!

File: assets/tests/components/detours/diversionPage.test.tsx (hunk 5 above)
Comment (hannahpurcell) [reply assets/tests/components/detours/diversionPage.test.tsx:275]: 🤔 Twas not a typo, but if this makes it meaningfully more readable to you, might as well fix.

File: assets/tests/components/detours/diversionPage.test.tsx (hunk 3 above)
Comment (firestack) [reply assets/tests/components/detours/diversionPage.test.tsx:302]:
```suggestion
    act(() => {
      fireEvent.click(originalRouteShape.get(container))
    })

    act(() => {
      fireEvent.click(originalRouteShape.get(container))
    })
```

File: assets/tests/components/detours/diversionPage.test.tsx (hunk 5 above)
Comment (joshlarson) [reply assets/tests/components/detours/diversionPage.test.tsx:275]: WELP!

I don't feel strongly about it, but I _think_ most of the test titles read more like full sentences?

Comment (hannahpurcell) [reply assets/tests/components/detours/diversionPage.test.tsx:275]: I'm not sure we can/should aim for unanimity in sentence structure across devs for test titles, but I'll commit this one since all I have to do is click a button!

File: assets/tests/components/detours/diversionPage.test.tsx [hunk 6]
```diff
@@ -242,6 +242,70 @@ describe("DiversionPage", () => {
...
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    fireEvent.click(screen.getByRole("button", { name: "Undo" }))
```
Comment (firestack) [inline assets/tests/components/detours/diversionPage.test.tsx:304]: nit(non-blocking): hmm, these seem to be undoing each-other, and we're not waiting on the results of the `Undo` click, could we remove both of these?

```suggestion
```

**Review by firestack — APPROVED** [review]
LGTM, Thanks @hannahpurcell!!

File: assets/tests/components/detours/diversionPage.test.tsx (hunk 6 above)
Comment (hannahpurcell) [reply assets/tests/components/detours/diversionPage.test.tsx:304]: Oh derp, embarrassing. My thought process was that I want to verify that the "regular route" message is absent in both situations where a detour is "unfinished" -- pre-completion of detour, and post-undo. I can tuck a `expect(await screen.findByText("Regular Route")).toBeVisible()` before the undo, but then we have repeated tests, essentially

Comment (joshlarson) [reply assets/tests/components/detours/diversionPage.test.tsx:304]: It _sort of_ feels like this is actually two tests hiding in one. First, testing that `Regular Route` doesn't appear before connecting a detour, and second, testing that `Regular Route` disappears after undoing a connected detour. 

Please treat ☝️ as musings, not things to act on right now, unless you want to :). When we come to a clearer testing strategy overall, that'll be a good opportunity to rework some of these tests.

Comment (hannahpurcell) [reply assets/tests/components/detours/diversionPage.test.tsx:304]: It's true, I did want to lump two tests together, heh. they are "similar vibe" but different things 😅
